app/controllers/application_controller.rb

```diff
@@ -32,6 +32,12 @@ class ApplicationController < ActionController::Base
     redirect_to main_app.root_url
   end
 
+  rescue_from ActiveRecord::RecordNotFound do |_exception|
+    flash[:error] = 'The object you are trying to look up does not exist '\
```

Comment:
Much better. I'm not 100% sure about the error message, how about something like "Oops, you tried to go somewhere that doesn't exist."? How does that sounds to you?